python/cuml/ensemble/randomforestclassifier.pyx

```diff
@@ -449,7 +450,12 @@ class RandomForestClassifier(BaseRandomForestModel,
         if self.random_state is None:
             seed_val = <uintptr_t>NULL
         else:
-            seed_val = <uintptr_t>self.random_state
+            if isinstance(self.random_state, np.random.RandomState):
+                seed_val = <uintptr_t>self.random_state.get_state()[1][0]
+            elif isinstance(self.random_state, cp.random.RandomState):
+                seed_val = <uintptr_t>self.random_state._rk_seed
+            else:
+                seed_val = <uintptr_t>self.random_state
```

Comment:
The drawback here is that even float random_state values are accepted and are forcibly converted to <uintptr_t>. (This is also happening in the original implementation outside of the PR). 
Is it better to raise a TypeError for all other types?